### PR TITLE
Introduction To Solid Principle

### PR DESCRIPTION
**Pull request by corpsgeek**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by corpsgeek** [comment]
This pull request is linked to: https://github.com/section-io/engineering-education/issues/903

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Comment by corpsgeek** [comment]
Alright.

File: articles/introduction-to-solid-principle/index.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+# What is Solid Principle?
+The SOLID principle is a design principle and pattern intended to make software designs more understandable, flexible, and maintainable. In the software development process, when creating an object ‒ a solid object or an abstract object, this object's design is essential, notably in deciding the accessibility scope.
+
+The object case scenario might confuse newbies in software development. Here's a better analogy; imagine if cups had their handles at the top rather than at the side. How accessible or comfortable would it be to use this cup (object)? But if the handle remains at the side, they remain readily accessible and comfortable to use.
+
+In software development, the decision on how accessible and flexible an object is should occur during this object's design phase. This decision-making process pre-software development ensures usability, simplicity, ease of implementation, and accessibility towards making software better than others.
```
Comment (lalith1403) [inline articles/introduction-to-solid-principle/index.md:11]: Some clarity on the second sentence is required. Rewrite the sentence into 2 smaller sentences.

File: articles/introduction-to-solid-principle/index.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+SOLID principles have been tested and trusted by software developers, and the underlying technique of this principle, when applied correctly, can aid in developing standard object-oriented software applications.
+
+# Benefits of SOLID Principle
+The following are the five advantages of the SOLID principle when incorporated in your software design process:
+
+## Accessibility
+The SOLID Principle ensures easy access and control to object entities. The integrity of stable object-oriented applications provides easy access to objects, eliminating the risks of unintended inheritance.
+
+## Ease of refactoring
+Softwares change over time. Therefore developers need to build applications while keeping in mind the possibility of future changes. Poorly structured software applications make it difficult to refactor, but there is effortlessness in refactoring your codebase with the SOLID principle.
```
Comment (lalith1403) [inline articles/introduction-to-solid-principle/index.md:22]: Software and code. We don't use softwares and codes.

File: articles/introduction-to-solid-principle/index.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
+# Introduction to Solid Principle
+Writing clean codes is one of the core precepts of software development. There are various software design approaches to ensure an understandable, flexible, and maintainable code base.
+
+This article explains one of the object-oriented programming design principles that assist you as a software developer towards writing clean codes, without narrowing its application to any programming language, but from a universal point of view towards understanding the concept of the SOLID principle.
+
+# What is Solid Principle?
+The SOLID principle is a design principle and pattern intended to make software designs more understandable, flexible, and maintainable. In the software development process, when creating an object ‒ a solid object or an abstract object, this object's design is essential, notably in deciding the accessibility scope.
```
Comment (lalith1403) [inline articles/introduction-to-solid-principle/index.md:7]: What is a solid object or an abstract object?

File: articles/introduction-to-solid-principle/index.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+ //calculate cost of trip fare
+ }
+public function calculateTotalCost() {
+ //use results from calculateFoodCost and calculateTripCost to return total cost
+ }
+}
+```
+The above code ensures that to write new costing criteria, we don't have to rewrite the costing criteria' functions, which is how you apply the open and close principle.
+
+## LISKOV Substitution Principle
```
Comment (lalith1403) [inline articles/introduction-to-solid-principle/index.md:140]: Use H3(###) and beyond. Change throughout the article wherever applicable.

File: articles/introduction-to-solid-principle/index.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+ //create Car logic and stores it using MySQLDatabaseManagement store method
+ }
+ 
+ public function delete(argument1) {
+ //delete a car from the database using MySQLDatabaseManagement the delete method
+ }
+}
+```
+
+Above, we applied the MySQLDatabaseManagement class for performing database operations, and it carries out its function. The problem with the code above is if we decide to change our database type from MySQL to Postgres.
```
Comment (lalith1403) [inline articles/introduction-to-solid-principle/index.md:214]: Use formatting to refer to variable names. `MySQLDatabaseManagement`

Comment (lalith1403) [reply articles/introduction-to-solid-principle/index.md:214]: Apply this accordingly throughout the article.

File: articles/introduction-to-solid-principle/index.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+ //create Car logic and stores it using DatabaseManagement store method
+ }
+ 
+ public function delete(argument1) {
+ //delete a car from the database using DatabasedManagement the delete method
+ }
+}
+```
+
+In the above implementation, the CarManufacturer class does not rely on a concrete class. Rather it relies on an interface. Therefore, if we intend to change the database technology, we can change the class that implements the DatabaseManagement interface.
```
Comment (lalith1403) [inline articles/introduction-to-solid-principle/index.md:245]: `CarManufacturer`

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Amazing article. I learned a lot about the SOLID principle and how to apply it. The examples are intuitive. Well done. A few minor changes and this should be good to go.

**Comment by corpsgeek** [comment]
Alright, I will work on it. Thank you.

**Comment by corpsgeek** [comment]
@lalith1403 revision complete.

**Comment by corpsgeek** [comment]
@lalith1403 drawing your attention here post weekend🤗

**Comment by hectorkambow** [comment]
Plagiarism check done 👍

**Comment by lalith1403** [comment]
The article is good to go. However, I found a few errors that I had resolved earlier. To avoid this, always pull the changes made before rectifying the errors.

**Review by lalith1403 — APPROVED** [review]

**Comment by corpsgeek** [comment]
> The article is good to go. However, I found a few errors that I had resolved earlier. To avoid this, always pull the changes made before rectifying the errors.

Alright, thank you.

**Comment by hectorkambow** [comment]
👋  Good evening @corpsgeek  

Just going over a final review for your article - it was very informative. 
Please be sure to go over the final edits I made to learn and carry over any applicable changes to future articles. 

There was a paragraph in there that was quite confusing, and I changed it altogether. 

I went ahead and made the changes to make the article flow - and I will be added a tiered payout label to this article (FYI). 

🚀 
cc @lalith1403

**Comment by corpsgeek** [comment]
Thank you @hectorkambow. I’ve gone through the changes.

**Pull request merged**